tokio-io/src/async_write.rs

```diff
@@ -143,12 +143,25 @@ pub trait AsyncWrite {
     }
 }
 
-/*
-impl<T: ?Sized + AsyncWrite> AsyncWrite for Box<T> {
-    fn shutdown(&mut self) -> Poll<(), std_io::Error> {
-        (**self).shutdown()
+impl<T: ?Sized + Unpin + AsyncWrite> AsyncWrite for Box<T> {
```

Comment:
crazy nit, but maybe the `AsyncX` and `Unpin` should be sorted in a consistent order?